src/main/java/com/parkhomovsky/bookstore/controller/BookController.java

```diff
@@ -49,7 +50,7 @@ public BookDto getById(@PathVariable Long id) {
     @GetMapping("/search")
     @Operation(summary = "Search books",
             description = "Search for books by title or author.")
-    public List<BookDto> search(BookSearchParameters parameters) {
+    public List<BookDto> search(@ModelAttribute BookSearchParameters parameters) {
```

Comment:
As far as I know, this annotation is needed for MVC applications. Do you really need this?